lib/apply-extends.js

```diff
@@ -0,0 +1,19 @@
+var fs = require('fs')
+var path = require('path')
+var assign = require('./assign')
+
+function applyExtends (config, cwd) {
+  var defaultConfig = {}
+
+  if (config.hasOwnProperty('extends')) {
+    var pathToDefault = path.join(cwd, config.extends)
+    delete config.extends
+
+    defaultConfig = JSON.parse(fs.readFileSync(pathToDefault, 'utf8'))
+    defaultConfig = applyExtends(defaultConfig, path.dirname(pathToDefault))
```

Comment:
[Question] I wonder if we should guard against circular references? It might be unlikely but would overflow the stack. Perhaps this submodule should keep track of the paths it has already loaded?